lib/toy/class_instance.rb

```diff
@@ -9,6 +9,12 @@ def superclass
       @superclass
     end
 
+    def ancestors
+      class_ancestors = super.uniq { |ancestor| ancestor.__id__ }
```

Comment:
As with the `#flat_map` below, you could do `uniq(&:__id__)` here if you wanted to.